pkg/gcc/rate_controller_buckets.go

```diff
@@ -105,18 +105,26 @@ func (c *rateControllerBuckets) onDelayStats(ds DelayStats) {
 	c.delayStats = ds
 	c.delayStats.State = c.delayStats.State.transition(ds.Usage)
 
-	if c.delayStats.State == stateHold {
-		c.bitrateControlBucketsManager.HandleBitrateNormal(uint64(c.target))
-		return
-	}
-
 	var next DelayStats
 
 	c.lock.Lock()
 
 	switch c.delayStats.State {
 	case stateHold:
-		// should never occur due to check above, but makes the linter happy
+		c.bitrateControlBucketsManager.HandleBitrateNormal(uint64(c.target))
```

Comment:
do we need to acquire the lock here?

Reply:
not really as nothing is being set. But, then, need to move around the code or duplicate the lock for each case.

Reply:
Forget this change, I will revert that part as no real point